### PR TITLE
Changes REST, spec descriptions for /supportMatrix and /supportMatrix/servers

### LINKED ISSUE #183: REST API: Human friendly descriptions to be returned by api/supportMatrix
DRAFT

* CONFIG - Configuration - externalize and manage your configuration parameters outside your microservices
* FAULT_TOLERANCE - Fault Tolerance - all about bulkheads, timeouts, circuit breakers, retries, etc. for your microservices
* JWT_AUTH - JWT Propagation - manage and propagate security across your microservices
* METRICS - Metrics - Gather and create operational and business measurements for your microservices
* HEALTH_CHECKS - Health Checks - ensure your microservices is up and running by enabling health checks
* OPEN_API - Open API - Generate OpenAPI-compliant API documentation for your microservices
* OPEN_TRACING - Open Tracing - trace the flow of requests as they traverse your microservices
* REST_CLIENT - Rest Client - Invoke RESTful services in a type-safe manner

@cealsair

### PR DESCRIPTION
**Pull request by Karm**
Fixes #183 

New top level attributes are added, "configs" and "descriptions".
"configs" contain what /supportMatrix/servers and /supportMatrix used to contain.
"descriptions" displays mapping between spec names and their legend.

Note that it changes already released API v1. I think it is kinda O.K. in our case as these specific resources are intended for [VSCode Extension](https://github.com/microprofile-starter-extensions/vscode-microprofile). If you don't like it, I will refactor it to API v2.

Note the MP specs are not sorted in this PR. Sorting PR is #186 

[See full examples on how it looks like now.](https://gist.github.com/Karm/e34fa6676b8bff43f70e3e9fa053cf16)

[REST-README.md](https://github.com/Karm/microprofile-starter/blob/issue-183/src/main/resources/REST-README.md#microprofile-versions-as-keys) was also updated.

### CODE REVIEW

File: src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java [hunk 1]
```diff
@@ -30,39 +30,49 @@
...
     CONFIG("config", "Config",
+            "Configuration - externalize and manage your configuration parameters outside your microservices",
             Arrays.asList(MicroProfileVersion.MP12, MicroProfileVersion.MP13, MicroProfileVersion.MP14,
                     MicroProfileVersion.MP20, MicroProfileVersion.MP21, MicroProfileVersion.MP22))
     , FAULT_TOLERANCE("fault_tolerance", "Fault Tolerance",
+            "Fault Tolerance - all about bulkheads, timeouts, circuit breakers, retries, etc. for your microservices",
             Arrays.asList(MicroProfileVersion.MP12, MicroProfileVersion.MP13, MicroProfileVersion.MP14,
                     MicroProfileVersion.MP20, MicroProfileVersion.MP21, MicroProfileVersion.MP22))
     , JWT_AUTH("JWT_auth", "JWT Auth",
+            "JWT Propagation - manage and propagate security across your microservices",
```
Comment (rdebusscher) [inline src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java]: Only propagate

File: src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java [hunk 2]
```diff
@@ -30,39 +30,49 @@
...
             Arrays.asList(MicroProfileVersion.MP12, MicroProfileVersion.MP13, MicroProfileVersion.MP14,
                     MicroProfileVersion.MP20, MicroProfileVersion.MP21, MicroProfileVersion.MP22))
     , METRICS("metrics", "Metrics",
+            "Metrics - Gather and create operational and business measurements for your microservices",
             Arrays.asList(MicroProfileVersion.MP12, MicroProfileVersion.MP13, MicroProfileVersion.MP14,
                     MicroProfileVersion.MP20, MicroProfileVersion.MP21, MicroProfileVersion.MP22))
     , HEALTH_CHECKS("health_checks", "Health Checks",
+            "Health Checks - ensure your microservices is up and running by enabling health checks",
```
Comment (rdebusscher) [inline src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java]: Health check is more than just is it up and running

Health Checks - Verify the health of your microservices with custom verifications

**Comment by rdebusscher** [comment]
Released API cannot be changed in a non backwards way. We can never be sure if someone is already relying on it.

So the /1 should stay and the new reply should be available under /2.

**Comment by Karm** [comment]
O.K.      Once @yeekangc  confirms that this is what he wants for the VSCode extension, I refactor it to appear only on latest and /2.    

 @yeekangc you can have a look already, the URL stays the same ```/api/```   (not specifying version means latest)

https://test-start.microprofile.io/api/supportMatrix/servers

https://test-start.microprofile.io/api/supportMatrix

Mind that if any PR is merged in the meantime, the current test site is refreshed with that...

File: src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java (hunk 1 above)
Comment (Karm) [reply src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java]: Addressed.

File: src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java (hunk 2 above)
Comment (Karm) [reply src/main/java/org/eclipse/microprofile/starter/addon/microprofile/servers/model/MicroprofileSpec.java]: Addressed.

**Comment by Karm** [comment]
@rdebusscher Redy for new review. 

* /api/2/supportMatrix and /api/2/supportMatrix/servers have the new format
* /api/supportMatrix and /api/supportMatrix/servers have the new format too (default matches latest as per our original design)
* /api/1/supportMatrix and /api/1/supportMatrix/servers retains the old format

**Comment by Karm** [comment]
@cealsair @rdebusscher What is holding this PR?

**Pull request merged**